pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -349,6 +361,10 @@ func (s *DiscoveryServer) handleCds(con *XdsConnection, discReq *xdsapi.Discover
 			errCode := codes.Code(discReq.ErrorDetail.Code)
 			adsLog.Warnf("ADS:CDS: ACK ERROR %s %s:%s", con.ConID, errCode.String(), discReq.ErrorDetail.GetMessage())
 			incrementXDSRejects(cdsReject, con.node.ID, errCode.String())
+			if s.InternalGen != nil {
```

Comment:
Alternative to all the `if != nil` is to make `nil.OnNack` do nothing instead of panic. Not sure if that is more or less confusing though

Reply:
I have a bit of a panic attack when I read things like nil.OnNack(). Many years of java and c.

Reply:
Yeah thats probably for the best, probably leads to more confusion than it is worth